site/cgi-bin/pdf_check.cgi

```diff
@@ -87,23 +86,29 @@ try:
     for filename in os.listdir(bulk_path):
 
         # recalculate the total # of pages for each file
-        total_pages = subprocess.check_output(["pdftk", filename, "dump_data"])
-        total_pages = total_pages.decode('utf-8').rstrip()
-        left_index = total_pages.find("NumberOfPages") + 15
-        right_index = total_pages.find("PageMediaBegin") - 1
-        total_pages = int(total_pages[left_index: right_index])
+        pdfFileObj = open(filename, 'rb')
+        pdfReader = PyPDF2.PdfFileReader(pdfFileObj)
+        total_pages = pdfReader.numPages
 
         div = total_pages // num
         
-        for j in range(0,div):
-            out_pdf = filename[:-4] + "_" + str(j) + ".pdf"
-            out_cover_pdf = filename[:-4] + "_" + str(j) + "_cover.pdf"
-            start = j*num+1
-            stop = (j+1)*num
-            subprocess.call(["pdftk", filename, "cat", str(start) + "-" + str(stop), "output", out_pdf])
-            subprocess.call(['pdftk', filename, 'cat', str(start), 'output', out_cover_pdf])
-    message += "=> finished pdftk"
-
+        # pdf = PdfFileReader(path)
+        counter = 0
+        for i in range(0, total_pages, num):
+            cover_writer = PdfFileWriter()
+            cover_writer.addPage(pdfReader.getPage(counter)) 
+            cover_filename = '{}_{}_cover.pdf'.format(filename[:-4], int(i/2))
+            pdf_writer = PdfFileWriter()
+            start = counter
+            for j in range(start, start+num):
+                pdf_writer.addPage(pdfReader.getPage(j)) 
+                output_filename = '{}_{}.pdf'.format(filename[:-4], int(i/2))
```

Comment:
Move this string naming outside the for loop, it doesn't change through the loop.